src/constructs/acm/certificate.ts

```diff
@@ -8,7 +8,8 @@ import type { GuStack } from "../core";
 import { AppIdentity } from "../core/identity";
 import type { GuMigratingResource } from "../core/migrating";
 
-export type GuCertificateProps = Record<Stage, GuDnsValidatedCertificateProps> & GuMigratingResource & AppIdentity;
+export type GuCertificateProps = Record<Stage, GuDnsValidatedCertificateProps> & GuMigratingResource;
+export type GuCertificatePropsWithApp = GuCertificateProps & AppIdentity;
```

Comment:
Feels a little horrible to make this split just to make the pattern props API simpler. The only alternative I can think of is to `GuEc2AppProps` to [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) the app field in `GuCertificateProps`, but not much better.

Might be one to address later once the AppIdentity mixin is written?

Reply:
> The only alternative I can think of is to `GuEc2AppProps` to [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) the app field in `GuCertificateProps`, but not much better.

Yep, I tried this alternative too and decided that the approach in this PR was 'least-bad' 🙈

Reply:
> Might be one to address later once the AppIdentity mixin is written?

I've already found myself needing to do this twice today ([see also](https://github.com/guardian/cdk/pull/446/files#diff-e4e7eb9118f3261284cbc3ed695b73b19661de3cbb7b438e102a8293a50719d1R19-R20)). So it's definitely going to be worth thinking about a general solution for this.